prime-angular-frontend/src/app/modules/enrolment/shared/services/enrolment-form-state.service.ts

```diff
@@ -398,6 +409,29 @@ export class EnrolmentFormStateService extends AbstractFormStateService<Enrolmen
     });
   }
 
+  private convertCareSettingFormToJson(enrolleeId: number): any {
+    // Variable names must match keys for FormArrays in the form to get values
+    const { careSettings, selectableHealthAuthorities } = this.careSettingsForm.getRawValue();
+
+    // Any checked HA is converted into an enrollee health authority,
+    // which is used to create the payload to back-end
+    let enrolleeHealthAuthorities = selectableHealthAuthorities.map((checkState, i) => {
```

Comment:
@neophyte57  alternatively if you want to avoid the filter when you see a map followed by a filter you can almost always replace it with a reduce:

```js
const {  careSettings, enrolleeHealthAuthorities: updatedSelections } = this.careSettingsForm.getRawValue();

const enrolleeHealthAuthorities = updatedSelections
  .reduce((selectedHealthAuthorities: ProperType, checked: boolean, i) => {
    if (checked) {
      selectedHealthAuthorities.push({ enrolleeId, healthAuthorityCode: this.configService.healthAuthorities[i] })
    }
    return selectedHealthAuthorities;
  }, []);

return {  careSettings,  enrolleeHealthAuthorities };
```

Reply:
@neophyte57 don't know if you saw the comment above, but instead of filtering out only a single negative result instead of all negative results we can avoid the filter and use a reduce.

Reply:
@mtpultz Sorry missed the `reduce` earlier

Reply:
Looks awesome @neophyte57